### PR TITLE
sdk 483 code sniffer

### PR DESCRIPTION
**Pull request by ScarIetSky**

### CODE REVIEW

File: tests/Sdk/Acceptance/Extension/Tasks/CodeSnifferTaskCest.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        $process = $I->runSdkCommand([
+            static::COMMAND,
+            '--path=src/CodeSniffer/fail',
+        ]);
+
+        // Assert
+        Assert::assertFalse($process->isSuccessful());
+        Assert::assertFileExists($I->getPathFromProjectRoot('reports/phpcs.codestyle.json'));
+        Assert::assertStringContainsString('Violation', $process->getOutput());
+        Assert::assertStringContainsString('Class name "Success" doesn\'t match filename, expected "Fail"', $process->getOutput());
```
Comment (miroshnikov-spryker) [inline tests/Sdk/Acceptance/Extension/Tasks/CodeSnifferTaskCest.php:66]: I don't think the exact error messages should be tested. The message comes from the underlying tool and when tested, a dependency on that tool is created. But we want to test the command, not a tool.

File: tests/_support/data/project/src/CodeSniffer/fail/Fail.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+namespace SprykerSdk\Sdk\Acceptance\Extension\Tasks\CodeSniffer;
+
+class Success
+{
+    /**
+     * @var array<string>
+     */
+    protected array $test;
```
Comment (Kyrylo-Meretskyi) [inline tests/_support/data/project/src/CodeSniffer/fail/Fail.php:15]: According to type `$test` should be renamed to `$tests`

File: tests/_support/data/project/src/CodeSniffer/success/Success.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+namespace SprykerSdk\Sdk\Acceptance\Extension\Tasks\CodeSniffer;
+
+class Success
+{
+    /**
+     * @var array<string>
+     */
+    protected array $test;
```
Comment (Kyrylo-Meretskyi) [inline tests/_support/data/project/src/CodeSniffer/success/Success.php:15]: According to type $test should be renamed to $tests

File: composer.json [hunk 4]
```diff
@@ -90,8 +90,8 @@
         "symfony/polyfill-php72": "*"
     },
     "scripts": {
-        "cs-check": "phpcs --colors -p -s --extensions=php --standard=vendor/spryker/code-sniffer/Spryker/ruleset.xml --ignore=/tests/_data/, src/ tests/ extension/",
-        "cs-fix": "phpcbf --colors -p --extensions=php --standard=vendor/spryker/code-sniffer/Spryker/ruleset.xml --ignore=/tests/_data/, src/ tests/ extension/",
+        "cs-check": "phpcs --colors -p -s --extensions=php --standard=vendor/spryker/code-sniffer/Spryker/ruleset.xml --ignore=/tests/_support/data/project/src/CodeSniffer/, src/ tests/ extension/",
+        "cs-fix": "phpcbf --colors -p --extensions=php --standard=vendor/spryker/code-sniffer/Spryker/ruleset.xml --ignore=/tests/_support/data/project/src/CodeSniffer/, src/ tests/ extension/",
```
Comment (Kyrylo-Meretskyi) [inline composer.json:94]:
```suggestion
        "cs-fix": "phpcbf --colors -p --extensions=php --standard=vendor/spryker/code-sniffer/Spryker/ruleset.xml --ignore=/tests/_support/data/, src/ tests/ extension/",
```

**Review by Kyrylo-Meretskyi — CHANGES_REQUESTED** [review]

File: tests/Sdk/Acceptance/Extension/Tasks/CodeSnifferTaskCest.php [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+        // Act
+        $process = $I->runSdkCommand([
+            static::COMMAND,
+            '--path=src/CodeSniffer/success',
+        ]);
+
+        // Assert
+        Assert::assertTrue($process->isSuccessful());
+        Assert::assertFileExists($I->getPathFromProjectRoot('reports/phpcs.codestyle.json'));
```
[the remark this answers is not in the document]
Comment (ScarIetSky) [reply tests/Sdk/Acceptance/Extension/Tasks/CodeSnifferTaskCest.php:38]: the name is hardcoded for this command.

File: tests/Sdk/Acceptance/Extension/Tasks/CodeSnifferTaskCest.php (hunk 1 above)
Comment (ScarIetSky) [reply tests/Sdk/Acceptance/Extension/Tasks/CodeSnifferTaskCest.php]: Yes, we can change the tool without changing the command. I agree.

File: tests/_support/data/project/src/CodeSniffer/fail/Fail.php (hunk 2 above)
Comment (ScarIetSky) [reply tests/_support/data/project/src/CodeSniffer/fail/Fail.php]: fixed

File: tests/_support/data/project/src/CodeSniffer/success/Success.php (hunk 3 above)
Comment (ScarIetSky) [reply tests/_support/data/project/src/CodeSniffer/success/Success.php]: fixed

File: tests/Sdk/Acceptance/Extension/Tasks/CodeSnifferTaskCest.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (miroshnikov-spryker) [reply tests/Sdk/Acceptance/Extension/Tasks/CodeSnifferTaskCest.php:38]: Do you mean this name: `phpcs.codestyle.json`? This file is not a violation report produced by the command but raw output of the tool. It is parsed by a ViolationReportConverter and may be presented in different ways according to `--format` option. If you pass `--format=yaml` the command should produce a report `reports/validation:php:codestyle-fix.violations.yaml`

**Review by Kyrylo-Meretskyi — APPROVED** [review]

File: tests/Sdk/Acceptance/Extension/Tasks/CodeSnifferTaskCest.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (ScarIetSky) [reply tests/Sdk/Acceptance/Extension/Tasks/CodeSnifferTaskCest.php]: ok. didn't know

**Review by miroshnikov-spryker — APPROVED** [review]

**Pull request merged**